bindings/pyroot_experimental/PyROOT/src/TTreePyz.cxx

```diff
@@ -236,3 +236,155 @@ PyObject *PyROOT::SetBranchAddressPyz(PyObject * /* self */, PyObject *args)
    // Not the overload we wanted to pythonize, return None
    Py_RETURN_NONE;
 }
+
+// Try ( const char*, void*, const char*, Int_t = 32000 )
+PyObject *TryBranchLeafListOverload(int argc, PyObject *args)
+{
+   PyObject *treeObj = nullptr;
+   PyObject *name = nullptr, *address = nullptr, *leaflist = nullptr, *bufsize = nullptr;
+
+   if (PyArg_ParseTuple(args, const_cast<char *>("OO!OO!|O!:Branch"),
+                        &treeObj,
+                        &CPyCppyy_PyUnicode_Type, &name,
+                        &address,
+                        &CPyCppyy_PyUnicode_Type, &leaflist,
+                        &PyInt_Type, &bufsize)) {
+
+      auto treeProxy = (CPPInstance *)treeObj;
+      TTree *tree = (TTree *)GetClass(treeProxy)->DynamicCast(TTree::Class(), treeProxy->GetObject());
+      if (!tree) {
+         PyErr_SetString(PyExc_TypeError, "TTree::Branch must be called with a TTree instance as first argument");
+         return nullptr;
+      }
+
+      void *buf = nullptr;
+      if (CPPInstance_Check(address))
+         buf = (void *)((CPPInstance *)address)->GetObject();
+      else
+         Utility::GetBuffer(address, '*', 1, buf, false);
+
+      if (buf) {
+         TBranch *branch = nullptr;
+         if (argc == 5) {
+            branch = tree->Branch(CPyCppyy_PyUnicode_AsString(name), buf, CPyCppyy_PyUnicode_AsString(leaflist),
+                                  PyInt_AS_LONG(bufsize));
+         } else {
+            branch = tree->Branch(CPyCppyy_PyUnicode_AsString(name), buf, CPyCppyy_PyUnicode_AsString(leaflist));
+         }
+
+         return BindCppObject(branch, Cppyy::GetScope("TBranch"));
+      }
+   }
+   PyErr_Clear();
+
+   Py_RETURN_NONE;
+}
+
+// Try ( const char*, const char*, T**, Int_t = 32000, Int_t = 99 )
+// or  ( const char*,              T**, Int_t = 32000, Int_t = 99 )
+PyObject *TryBranchPtrToPtrOverloads(int argc, PyObject *args)
+{
+   PyObject *treeObj = nullptr;
+   PyObject *name = nullptr, *clName = nullptr, *address = nullptr, *bufsize = nullptr, *splitlevel = nullptr;
+
+   auto bIsMatch = false;
+   if (PyArg_ParseTuple(args, const_cast<char *>("OO!O!O|O!O!:Branch"),
+                        &treeObj,
+                        &CPyCppyy_PyUnicode_Type, &name,
+                        &CPyCppyy_PyUnicode_Type, &clName,
+                        &address,
+                        &PyInt_Type, &bufsize,
+                        &PyInt_Type, &splitlevel)) {
+      bIsMatch = true;
+   } else {
+      PyErr_Clear();
+      if (PyArg_ParseTuple(args, const_cast<char *>("OO!O|O!O!"),
+                           &treeObj,
+                           &CPyCppyy_PyUnicode_Type, &name,
+                           &address,
+                           &PyInt_Type, &bufsize,
+                           &PyInt_Type, &splitlevel)) {
+         bIsMatch = true;
+      } else
```

Comment:
is it good for our coding conventions to skip braces?

Reply:
Done!